test/tests/functional/pbs_calendaring.py

```diff
@@ -97,7 +97,43 @@ def test_topjob_start_time(self):
         # since only one subjob of array parent can become topjob
         # second job must start 10 seconds after that because
         # walltime of array job is 10 seconds.
-        self.assertEqual(est_epoch2, est_epoch1 + 10)
+        self.assertEqual(est_epoch2, est_epoch1 + 30)
         # Also make sure that since second subjob from array is running
         # Third subjob should set estimated.start_time in future.
         self.assertGreater(est_epoch1, time_now)
+
+    def test_topjob_start_time_of_subjob(self):
+        """
+        In this test we test that the subjob which gets added to the
+        calendar as top job and it has estimated start time correctly set when
+        job history is enabled and opt_backfill_fuzzy is turned off.
+        """
+
+        self.scheduler.set_sched_config({'strict_ordering': 'true all'})
+        a = {'resources_available.ncpus': 1}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+        a = {'backfill_depth': '2', 'job_history_enable': 'True'}
```

Comment:
While it doesn't harm anything, you don't need to turn on job history.  While the job array is still around, subjobs will be in the X state.  If job history is enabled, all subjobs will go into the F state when the job array finishes.  If you remove this, consider removing it from other tests as well (like the one above this one).

Reply:
That was a copy paste error, it is not needed.
For the first test, when it was written we were facing a problem with history jobs because server used to report expired subjobs to the scheduler and it used to cause trouble in finding the start time of the subjobs. I think it is ok to have it there.